sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -690,6 +720,35 @@ def decode_from_stream(self, in_stream, nested):
           elements.append(self._elem_coder.decode_from_stream(in_stream, True))
         count = in_stream.read_var_int64()
 
+      if count == -1:
+        if self._read_state is None:
+          raise ValueError(
+              'Cannot read state-written iterable without state reader.')
+
+        class FullIterable(object):
+          def __init__(self, head, tail):
+            self._head = head
+            self._tail = tail
+
+          def __iter__(self):
+            for elem in self._head:
+              yield elem
+            for elem in self._tail:
+              yield elem
+
+          def __eq__(self, other):
+            return list(self) == list(other)
+
+          def __hash__(self):
+            raise NotImplementedError
+
+          def __reduce__(self):
+            return list, (list(self),)
+
+        state_token = in_stream.read_all(True)
```

Comment:
shouldn't this read the length of the state token from in_stream?

Reply:
Yes, that's what the `read_all([nested=]True)` does.